### PR TITLE
idbfs: don't ignore changes files if timestamp decreased

### PR DESCRIPTION
**Pull request by Beuc**
This saves a file not only if the file's timestamp was increased due to a direct change, but also when the timestamp decreased due to e.g. an in-process unzip which restored a file with an earlier timestamp.

(Initiating pull request following #12348 -- cf. discussion there)

### CODE REVIEW

**Review by kripken — APPROVED** [review]
I agree this should be safe, despite a very minor risk. Let's see if others disagree though.

Please add a mention to the Changelog for this, may help someone someday even if this is not likely to hurt anyone.

File: ChangeLog.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ Current Trunk
   version of clang rather than replacing it.
 - ASAN_SHADOW_SIZE is deprecated. When using AddressSanitizer, the correct
   amount of shadow memory will now be calculated automatically.
+- IDBFS now persists files whenever their timestamp changes; previously it acted
+  on sync only if the timestamp increased and ignored the file changes otherwise.
```
Comment (kripken) [inline ChangeLog.md:29]: we tagged 2.0.6 meanwhile, so this should go in `Current Trunk`.

Comment (kripken) [reply ChangeLog.md:29]: (side note, adding to the top of the list avoids this being a silent error, but then you'd get a merge conflict - so you still need to do a manual operation here, no way around that sadly...)

Comment (Beuc) [reply ChangeLog.md]: Done.

**Comment by Beuc** [comment]
Thanks!

**Comment by VinnyOG** [comment]
That `!=` doesn't check value equivalence like the comparison operators...
That checks object reference equivalence and always resolves to false in this case.
`new Date(1) != new Date(1)` even though new Date(1) > new Date(0)`

This is a bug

File: src/library_idbfs.js [hunk 2]
```diff
@@ -231,7 +231,7 @@ mergeInto(LibraryManager.library, {
       Object.keys(src.entries).forEach(function (key) {
         var e = src.entries[key];
         var e2 = dst.entries[key];
-        if (!e2 || e['timestamp'] > e2['timestamp']) {
+        if (!e2 || e['timestamp'] != e2['timestamp']) {
```
Comment (VinnyOG) [inline src/library_idbfs.js:234]: This compares object references and not actual date values. It always evaluates to `true`.

**Pull request merged**